### ISSUE TITLE
weird 403 response when not using hostname

### ISSUE THREAD

**Issue by zalmoxis2019** [warmcat/libwebsockets#1932]
Hello,

  I'm sending some http1.1 requests to a server and noticed that if i use the ip returned by lws_get_peer_simple(wsi, mco->dest_ip, 16);  for the next requests that i send. Sometimes the server replies with a 403 (forbidden) and sometimes it works perfectly getting 200 as a reply. After playing with all the possible client_info ssl flags:

new_request->client_info.ssl_connection = LCCSCF_USE_SSL | LCCSCF_PIPELINE | LCCSCF_ALLOW_SELFSIGNED | LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK | LCCSCF_ALLOW_EXPIRED | LCCSCF_ALLOW_INSECURE;

 and noticing that this behavior keeps happening, i thought it's some issue with the server and not on my side (client side). But then i used another library to send my http requests (POCO) and this issue doesn't happen at all. 

   Have you seen this issue before? Can you please point me please towards something else i can test/experiment to figure out why this is happening?
Basically all is great if i pass the hostname instead of an ip... this is very weird.
new_request->client_info.host = new_request->client_info.address = last_request->dest_ip; //hostname;

thank you very much!

**Comment by lws-team** [comment]
If the server wants to send you a 403 or act in any way really, why it chose to do that is up to the server.

> Basically all is great if i pass the hostname instead of an ip... this is very weird.

Dump what you are sending and confirm what you are actually sending, not what you think you are sending.

https://libwebsockets.org/git/libwebsockets/tree/lib/core-net/output.c#n38-39

**Comment by zalmoxis2019** [comment]
I don't think it's a server issue because if i send the same thing with poco or curl i don't see this behavior at all. I also experimented to see what exactly i send using a http dump service like "www.ptsv2.com" and checked my lws logs too. And from the perspective of the content sent they are identical. (same headers, same body)
   I suspect this is about the ssl_connection flags but not sure how to identify what exactly that is.

**Comment by lws-team** [comment]
> I don't think it's a server issue 

I did not say it's a "server issue".  I said the server alone has the logic to decide whether what it sees makes it issue a 403.  The server is opaque and we can't look at its logs to find out what triggered it to do that.

> if i send the same thing with poco or curl i 

That's great, use those.

Otherwise if you take my time asking for advice, please don't argue with it wasting both our time, just follow what I suggested or close the issue.
